modules/odr_api/odr_api/api/endpoints/auth.py

```diff
@@ -46,7 +68,139 @@ def logout(db=Depends(get_db), session_cookie: str = Depends(get_session_cookie)
 
 
 @router.get("/auth/logout/all")
-def logout_all(db=Depends(get_db), current_user: User = Depends(AuthProvider())):
+def logout_all(current_user: User = Depends(AuthProvider()), db: Session = Depends(get_db)):
     user_crud.delete_user_sessions(db, current_user.id)
 
     return UserLogout()
+
+
+def ouath2_login_and_signup(
+    request: Request,
+    openid_user: OpenID,
+    db: Session
+):
+    if not openid_user.email:
+        raise HTTPException(status_code=400, detail="Unable to get user email")
+
+    existing_user = user_crud.get_user_by_email(db, openid_user.email)
+    if not existing_user:
+        existing_user = user_crud.create_user_from_openid(db, openid_user)
+
+    session = user_crud.login_openid_user(db, openid_user)
+    if not session:
+        raise HTTPException(status_code=401, detail="Different provider already used")
+
+    response = RedirectResponse(url=f"{request.base_url}{settings.OAUTH2_REDIRECT_PATH}")
+    response.set_cookie(key="session", value=session.id, httponly=True)
+
+    return response
+
+
+# Google SSO
+google_sso = GoogleSSO(
+    client_id=settings.GOOGLE_CLIENT_ID,
+    client_secret=settings.GOOGLE_CLIENT_SECRET,
+    redirect_uri="http://localhost:31100/api/v1/auth/google/callback",
```

Comment:
This is hardcoded as localhost, should it be?

Reply:
Should not be. Will break inside of the containers.

Reply:
Removed this property as it was not needed. Redirect uri is already set in different place.